proteuslib/tests/test_recarbonation_process.py

```diff
@@ -0,0 +1,541 @@
+
+    @pytest.mark.unit
+    def test_units_equilibrium(self, equilibrium_config):
+        model = equilibrium_config
+        assert_units_consistent(model)
+
+    @pytest.mark.unit
+    def test_dof_equilibrium(self, equilibrium_config):
+        model = equilibrium_config
+        assert (degrees_of_freedom(model) == 0)
+
+    @pytest.mark.component
+    def test_scaling_equilibrium(self, equilibrium_config):
+        model = equilibrium_config
+        iscale.calculate_scaling_factors(model.fs.unit)
+
+        assert hasattr(model.fs.unit.control_volume, 'scaling_factor')
+        assert isinstance(model.fs.unit.control_volume.scaling_factor, Suffix)
+
+        assert hasattr(model.fs.unit.control_volume.properties_out[0.0], 'scaling_factor')
+        assert isinstance(model.fs.unit.control_volume.properties_out[0.0].scaling_factor, Suffix)
+
+        assert hasattr(model.fs.unit.control_volume.properties_in[0.0], 'scaling_factor')
+        assert isinstance(model.fs.unit.control_volume.properties_in[0.0].scaling_factor, Suffix)
+
+        # When using equilibrium reactions, there are another set of scaling factors calculated
+        assert hasattr(model.fs.unit.control_volume.reactions[0.0], 'scaling_factor')
+        assert isinstance(model.fs.unit.control_volume.reactions[0.0].scaling_factor, Suffix)
+
+    @pytest.mark.component
+    def test_initialize_solver(self, equilibrium_config):
+        model = equilibrium_config
+        solver.options['bound_push'] = 1e-10
+        solver.options['mu_init'] = 1e-6
+        model.fs.unit.initialize(optarg=solver.options)
+        assert degrees_of_freedom(model) == 0
+
+    @pytest.mark.component
+    def test_solve_equilibrium(self, equilibrium_config):
+        model = equilibrium_config
+        solver.options['max_iter'] = 100
+        results = solver.solve(model)
+        print(results.solver.termination_condition)
+        assert results.solver.termination_condition == TerminationCondition.optimal
+        assert results.solver.status == SolverStatus.ok
+
+    @pytest.mark.component
+    def test_solution_equilibrium(self, equilibrium_config):
+        model = equilibrium_config
+
+        assert pytest.approx(300, rel=1e-3) == value(model.fs.unit.outlet.temperature[0])
+        assert pytest.approx(10, rel=1e-3) == value(model.fs.unit.outlet.flow_mol[0])
+        assert pytest.approx(101325, rel=1e-3) == value(model.fs.unit.outlet.pressure[0])
+
+        total_molar_density = \
+            value(model.fs.unit.control_volume.properties_out[0.0].dens_mol_phase['Liq'])/1000
+        assert pytest.approx(55.2336, rel=1e-3) == total_molar_density
+        pH = -value(log10(model.fs.unit.outlet.mole_frac_comp[0, "H_+"]*total_molar_density))
+        pOH = -value(log10(model.fs.unit.outlet.mole_frac_comp[0, "OH_-"]*total_molar_density))
+        assert pytest.approx(5.336, rel=1e-3) == pH
+        assert pytest.approx(8.598, rel=1e-3) == pOH
```

Comment:
I would add one more assertion here for the amount of CO2 absorbed into the liquid phase. We know from literature that our aqueous CO2 should be around 0.02754 mol/L at this given inlet molefraction, and since we are utilizing the built in 'fugacity' calculations of IDAES, we should check the solution here to make sure we get it correct. That way if anything changes downstream in IDAES, the test would catch it here. 

So, I would add the following lines...

CO2_sorbed = model.fs.unit.control_volume.properties_out[0.0].conc_mol_phase_comp[('Liq', 'CO2')].value
assert pytest.approx(27.54, rel=1e-3) == CO2_sorbed

(NOTE: The last decimal places may vary slightly)

Reply:
I have added the assertion for CO2 absorbed in liquid phase.